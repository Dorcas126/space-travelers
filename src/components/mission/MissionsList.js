import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Table } from 'react-bootstrap';
import { getMissions } from '../../redux/mission/missions';
import MissionItem from './MissionItem';

const MissionsList = () => {
  const dispatch = useDispatch();
  const missionsList = useSelector((state) => state.missions);

  useEffect(() => {
    dispatch(getMissions());
  }, [dispatch]);

  return (
    <Table striped bordered responsive>
      <thead>
        <tr>
          <th className="fw-bold">Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th className="text-center">Join/Leave</th>
        </tr>
      </thead>
      <tbody>
        {missionsList.map((mission) => (
          <MissionItem key={mission.mission_id} mission={mission} />
        ))}
      </tbody>
    </Table>
  );
};

export default MissionsList;
