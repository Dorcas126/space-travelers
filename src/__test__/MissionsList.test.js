import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import MissionsList from '../components/mission/MissionsList';

const mockStore = configureMockStore();
const store = mockStore({ missions: [] });

describe('MissionsList', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MissionsList />
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
