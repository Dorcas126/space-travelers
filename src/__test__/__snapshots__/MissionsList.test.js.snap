// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionsList renders correctly 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table table-striped table-bordered"
  >
    <thead>
      <tr>
        <th
          className="fw-bold"
        >
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th
          className="text-center"
        >
          Join/Leave
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
